source/NotionsOfDecidability/Decidable.lagda

```diff
@@ -403,3 +402,31 @@ module propositional-truncation-of-decidable-type
   decidable-types-have-split-support s = ∣∣⟨ δ ⟩-exit (∥∥-to-∥∥⟨_⟩ s)
 
 \end{code}
+
+Added by Fredrik Bakke on 26 March 2025.
+
+We define the concept of decidable maps as maps where each fiber is decidable.
+
+\begin{code}
+
+is-decidable-map : {X : 𝓤 ̇ } {Y : 𝓥 ̇ } → (X → Y) → 𝓤 ⊔ 𝓥 ̇
+is-decidable-map f = each-fiber-of f is-decidable
+
+∘-decidable-map : {X : 𝓤 ̇ } {Y : 𝓥 ̇ } {Z : 𝓦 ̇ } (f : X → Y) (g : Y → Z)
+                → left-cancellable g
+                → is-decidable-map g
+                → is-decidable-map f
+                → is-decidable-map (g ∘ f)
+∘-decidable-map f g H G F x = cases positive-case negative-case (G x)
+ where
+  positive-case : (u : fiber g x) → is-decidable (fiber (g ∘ f) x)
+  positive-case u =
```

Comment:
Avoid projections with the use of pattern matching, to try to make the code more readable. There are more places below where this should also be done.